eo-parser/src/main/java/org/eolang/parser/StHash.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.parser;
+
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import com.yegor256.xsline.StEnvelope;
+import com.yegor256.xsline.StLambda;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.xembly.Directives;
+import org.xembly.Xembler;
+
+/**
+ * Add attribute 'hash' in node 'program' in XML.
+ * Returns already XML with hash.
+ *
+ * @since 0.35.0
+ */
+public final class StHash extends StEnvelope {
```

Comment:
@Yanich96 I have some doubts about the `StHash` implementation and if we need it. I believe you can set the `hash` function directly in `XeEoListener` during the parsing. What do you think?

Reply:
@volodya-lombrozo Hash code computes from node 'objects'. There isn't this node in parsing stage.

Reply:
@volodya-lombrozo Xembly directives will have to be run 2 times: 1 - to calculate the hash, 2 - to completely parse, and during the parsing process there may be XSD schema errors and parsing errors, as a result of which the hash calculation ceases to make sense.

Reply:
> @volodya-lombrozo Hash code computes from node 'objects'. There isn't this node in parsing stage.

@Yanich96 Would `exitProgram` help us here? It looks like the final parsing stage.

Reply:
> @volodya-lombrozo Xembly directives will have to be run 2 times: 1 - to calculate the hash, 2 - to completely parse, and during the parsing process there may be XSD schema errors and parsing errors, as a result of which the hash calculation ceases to make sense.

We anyway will need to read XML twice

Reply:
@volodya-lombrozo I will write todo for it.